### PR TITLE
FeeManager: Fix SQL in accumulateFees

### PR DESCRIPTION
**Pull request by omerfirmak**
Old query had a syntax error. Also added a test for the DB storage.

### CODE REVIEW

File: source/agora/consensus/Fee.d [hunk 1]
```diff
@@ -515,3 +515,61 @@ public class FeeManager
         assert(tot_fee == Amount(0));
     }
 }
+
+unittest
+{
+    import agora.consensus.state.UTXOSet;
+    import agora.common.Hash;
+
+    const db_file = "fee_man.tmp";
```
Comment (AndrejMitrovic) [inline source/agora/consensus/Fee.d]: No IO in unittests please.

File: source/agora/consensus/Fee.d [hunk 2]
```diff
@@ -515,3 +515,61 @@ public class FeeManager
...
+    auto fee_man = new FeeManager(db_file, new immutable(ConsensusParams));
```
Comment (Geod24) [inline source/agora/consensus/Fee.d]: You can just introduce a new `private` constructor which takes a database

Comment (omerfirmak) [reply source/agora/consensus/Fee.d]: Yeah, will do that

File: source/agora/consensus/Fee.d [hunk 3]
```diff
@@ -370,9 +370,9 @@ public class FeeManager
             {
                 auto new_fee = this.accumulated_fees[stake.output.address]
                     .toString();
-                this.db.execute("INSERT INTO accumulated_fees (public_key, fee)"
-                    ~ " VALUES (?,?) ON CONFLICT(public_key) DO UPDATE SET fee = ?",
-                    stake.output.address.toString(), new_fee, new_fee);
+                this.db.execute("INSERT OR REPLACE INTO accumulated_fees " ~
```
Comment (linked0) [inline source/agora/consensus/Fee.d]: Couldn't `REPLACE INTO` be used instead of `INSERT OR REPLACE INTO`?

**Comment by linked0** [comment]
Please leave a description of this PR and add the message about why the change is made in commits.

**Comment by omerfirmak** [comment]
All done

**Review by AndrejMitrovic — APPROVED** [review]

**Pull request merged**